frontend/src/middlewares/error-handling/error-handling-middleware.ts

```diff
@@ -0,0 +1,22 @@
+import { type Middleware, type PayloadAction } from '@reduxjs/toolkit';
+import { toast } from 'react-toastify';
+
+type RejectedAction = PayloadAction<
+  unknown,
+  string,
+  unknown,
+  Error | undefined
+>;
+
+const handleAction =
+  (next: (action: RejectedAction) => RejectedAction) =>
+  (action: RejectedAction): RejectedAction => {
```

Comment:
Use AnyAction from redux-toolkit instead of rejected action
```suggestion
  (action: AnyAction): unknown => {
```